src/NLog/Config/XmlLoggingConfiguration.cs

```diff
@@ -321,28 +321,21 @@ private void Initialize(XmlReader reader, string fileName, bool ignoreErrors)
             catch (Exception exception)
             {
                 InitializeSucceeded = false;
-                if (exception.MustBeRethrown())
+                if (exception.MustBeRethrownImmediately())
                 {
                     throw;
                 }
 
-                NLogConfigurationException ConfigException = new NLogConfigurationException("Exception occurred when loading configuration from " + fileName, exception);
+                InternalLogger.Error(new NLogConfigurationException("Exception occurred when loading configuration from " + fileName, exception), "Error in Parsing Configuration File.");
 
                 if (!ignoreErrors)
                 {
-                    if (LogManager.ThrowExceptions)
-                    {
-                        throw ConfigException;
-                    }
-                    else
+                    if (exception.MustBeRethrown())
```

Comment:
The `InternalLogger.Error()` call above logs a new exception object, so `exception` is not marked as logged. Hence, `exception` will be logged here. Is this what we want? Must we not call `MustBeRethrown()` on the new exception object?